internal/ingress/store/store.go

```diff
@@ -290,7 +301,9 @@ func (s Store) ListKongCredentials() []*configurationv1.KongCredential {
 	var credentials []*configurationv1.KongCredential
 	for _, item := range s.stores.Credential.List() {
 		c, ok := item.(*configurationv1.KongCredential)
-		if ok && s.isValidIngresClass(&c.ObjectMeta) {
+		// TODO arguably we can just remove the second clause, as this always returns true
+		// ListCACerts is like this already
+		if ok && s.isValidIngressClass(&c.ObjectMeta, annotations.IgnoreClassMatch) {
```

Comment:
Why not remove the second clause now?

Reply:
These ignore listers are kinda the odd ones out. `ListCACerts` simply doesn't call `isValidIngressClass` at present, but that won't be the case going forward (since we're changing its behavior to require the annotation).

`ListKongCredentials` will remain ignore, but is weird because it's the only thing that ignores them, and is for a deprecated resource type. The other things that ignore class (e.g. KongPlugin) don't have their own `ListWhatever` functions because they're instead pulled in by the parser.

I kind of like leaving the `isValidIngressClass` call in here just because it makes the `ListWhatever` functions consistent: even though it's essentially `if ok && true` it's kind of a self-documenting code thing to explicitly say "we're validating the ingress class by ignoring it" rather than assuming you know there's no such call because it's ignored. This would be equally valid, however:

```
if ok // normally we'd call isValidIngressClass here, but we don't because we don't care about credential class
```

`KongCredential` is really odd because it's a "standalone" entity that we list and then combine with KongConsumers via its `consumerRef` field, but is (a) deprecated (most configuration should instead use a KongConsumer that references credential secrets, which in turn _don't_ link back to anything and have no `ListWhatever` function) and (b) references something that may or may not have a class annotation of its own (KongConsumer defaults to not requiring any at present).

Reply:
Not my style - to me, simpler logic means fewer bugs, and the documentation could be achieved by a brief comment like this:

```go
// ListKongCredentials disregards IngressClass.
if ok {
  ...
}
```

I won't block you on that, though.

Reply:
Leaving as-is then. All said this particular lister function is likely going away in the not-too-distant future, so I'm not too concerned with it being perfect.